xo-game/src/Components/Board/Board.jsx

```diff
@@ -1,22 +1,38 @@
-import React, { useCallback } from "react";
+import React, { useCallback, useState } from "react";
 import { useLocation, useNavigate } from "react-router-dom";
 import Xicon from "../icons/Xicon";
 import Oicon from "../icons/Oicon";
 import "./board.css";
 import BoardCard from "./BoardCard";
+import RestartPoupup from "../BoardPoupup/RestartPoupup";
 
 const Board = () => {
   const location = useLocation();
   const navigate = useNavigate();
   const players = location.state?.players || [];
-  const squares = ["", "O", "X", "", "O", "X", "", "", ""];
+  const [squares, setSquares] = useState(new Array(9).fill(""));
+  const [isRestartPopupOpen, setIsRestartPopupOpen] = useState(false);
+  const [xnext, setXnext] = useState(true);
 
   const handleGoBack = () => {
     navigate(-1);
   };
-  const handleReload = useCallback(() => {
-    window.location.reload();
-  }, []);
+  const handleToggleRestartPopup = () => {
+    setIsRestartPopupOpen(!isRestartPopupOpen);
+  };
+  const handleToggleRestartPopupCallback = useCallback(() => {
+    setSquares(new Array(9).fill(""));
+    setXnext(true)
+    handleToggleRestartPopup();
+  }, [handleToggleRestartPopup]);
```

Comment:
it's right to put function in dependency array but sometimes it leades to unnecessary r‎e-renders and performance issues, I prefer to put the state  (isRestartPopupOpen) not a function in the dependency array